app/src/main/java/com/asfoundation/wallet/repository/BackendTransactionRepository.kt

```diff
@@ -75,7 +75,19 @@ class BackendTransactionRepository(
           if (isRevertTransaction(transaction.type)) {
             localRepository.getRevertedTransaction(wallet, transaction.transactionId)
                 .map { link -> mapper.map(transaction, link) }
-                .onErrorReturn { mapper.map(transaction) }
+                .onErrorResumeNext {
+                  localRepository.getRevertedTxId(wallet, transaction.transactionId)
+                      .flatMap { link ->
+                        offChainTransactions.getTransactionsById(wallet, link)
+                            .map { result -> result[link] }
+                            .map { revertedTx -> transactionsMapper.map(revertedTx, wallet) }
+                            .doOnSuccess { revertedTx ->
```

Comment:
Here, you mapped the revertedTx, but on the doOnSuccess you still have a variable called revertedTx.